### ISSUE TITLE
CheckRawSocketPermissions() always fails for IPv6 on Unix and we fall back to executing ping6 utility as new process.

### ISSUE THREAD

**Issue by wfurt** [dotnet/corefx#30169]
Consider simple "ping" program:

```c#
using System.Net.NetworkInformation;

namespace Test
{
    class Program
    {
        static void Main(string[] args)
        {            
            using (Ping pinger = new Ping())
            {
                string dst = "2607:f8b0:400a:800::200e";
                PingReply reply = pinger.SendPingAsync(dst, 5000, new byte[] { 0xff }, new PingOptions(3, false)).Result;
                Console.WriteLine($"status: {reply.Status} address: {reply.Address}");
                Console.ReadLine();
            }
       }
    }
}
```

```
$ strace -f -t -e network ~/latest/dotnet run
[pid  7527] 08:49:30 socket(PF_INET6, SOCK_RAW, IPPROTO_ICMPV6) = -1 EPERM (Operation not permitted)
```

This is expected. We try to open RAW socket and if that fails we will do ping by starting new process and using system binary. 

Now, try same command with sufficient priviledge

```
$ sudo strace -f -t -e network ~/latest/dotnet run
[pid  7822] 08:56:05 socket(PF_INET6, SOCK_RAW|SOCK_CLOEXEC, IPPROTO_ICMP) = 28
[pid  7822] 08:56:05 setsockopt(28, SOL_IPV6, IPV6_V6ONLY, [1], 4) = -1 EINVAL (Invalid argument)
```

**Comment by kasper3** [comment]
`sudo` is repeated twice in: `sudo sudo strace -f -t -e network ~/latest/dotnet run`. is it intentional?

**Comment by wfurt** [comment]
That was cut & paste error @kasper3. The point was that the function does not fail on permissions. 
I bump to this and #30168 while working on fix for #9350.
It should be easy to fix IMHO.

**Comment by karelz** [comment]
What is impact of the bug? Does it affect only `Ping.SendPingAsync` or more?

**Comment by wfurt** [comment]
We always fail to executing process when using ping to IPv6 destination. 
I did not find references to this beyond tests and Ping. 
However the issue lives in Sockets so it seems all 

```c#
Socket s = new Socket(AddressFamily.InterNetworkV6,  SocketType.Raw, XXX);
```
will throw exception. The exposure is probably low as use of RAW sockets is rare IMHO.

**Comment by wfurt** [comment]
fixed by #30202
